### PR TITLE
Added support for allowEmptyValue in string type

### PR DESCRIPTION
**Pull request by qtomasicchio**
@Adrien-Martins @jdassonvil @Steph0

### CODE REVIEW

File: test/test-enjoi.js [hunk 1]
```diff
@@ -292,6 +292,28 @@ Test('types', function (t) {
...
+
+        Joi.validate('', schemaAllowingEmpty, function (error, value) {
+            t.ok(!error,  "Empty string allowed");
+        });
+
+        var defaultStringSchema = Enjoi({
+            'type': 'string'
+        });
+
+        Joi.validate('', defaultStringSchema, function (error, value) {
```
Comment (Steph0) [inline test/test-enjoi.js:323]: I would add also a test with property 'allowEmptyValue' explicitely to false to verify that it works like default behavior.

Comment (qtomasicchio) [inline test/test-enjoi.js:323]: Done. Added test for default behaviour with values: `false`, `'invalid'` and `1`.

**Comment by Marsup** [comment]
Is there any valid reason why this PR is not made against the original repository ?

**Comment by qtomasicchio** [comment]
Mainly the project deadline, customer review is next week. The original repo can be updated later.

**Comment by Steph0** [comment]
:+1:

**Comment by qtomasicchio** [comment]
As discussed with Marsup, `minLength` already cover this use case (see [enjoi.js#L220-L222](https://github.com/tlivings/enjoi/blob/master/lib/enjoi.js#L220-L222)). 
PR closed !

**Pull request closed without merging**